doc/tutorials/gapi/face_beautification/face_beautification.markdown

```diff
@@ -0,0 +1,500 @@
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp comp_str_1
+...
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp comp_str_2
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp comp_str_3
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp comp_str_4
+
+### 2) Frames processing {#gapi_fb_s_2}
+
+The next step used to be a videostream definition and a source setup:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_old.cpp header
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_old.cpp in_out_1
+
+and then a compiled graph used to be applied to every single frame.
+This is an example of a processing loop without Streaming:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_old.cpp in_out_2
+
+Now this can be done without explicit use of additional libraries. First of all,
+let's define a stream variable and pass to it:
+    - video properties (this unconvenience is temporary);
+    - graph compile arguments (which will be passed to the graph inside
+the stream):
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp str_comp
+
+Then we should set a source:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp str_src
+
+And for this moment our stream is ready to be launched! An example of a
+Streaming loop:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp str_loop
+
+There are some other useful possibilities provided, e.g. you can abort stream
+processing via `cv::GStreamingCompiled::stop()` method (the next call of
+`cv::GStreamingCompiled::start()` will start processing the stream from the very
+beginning).
+
+### 3) You are breathtaking! {#gapi_fb_s_3}
+
+Now, when all the new G-API features are described and understood, we can
+discuss some algorithms.
+
+# Face Beautification algorithm {#gapi_fb_algorithm}
+
+The algorithm of a face retouching has been chosen as a subject for this
+tutorial to discover G-API usage convenience and efficiency in the context of
+solving a real CV problem. Its main idea is to implement a real-time videostream
+processing that detects faces and applies some filters to make them look
+beautiful (more or less). The pipeline is the following:
+
+![The Face Beautification algorithm](pics/algo.png)
+
+The algorithm consists of two parts: an inference of two networks (including
+data pre- and post-processing) and an image filtering pipeline which uses the
+inference data (creating masks, applying filters and composing the output
+image).
+
+Details of obvious operations (such as command line parsing or a single-string
+kernel implementation) are not going to be described in this text; there will be
+only comments on debatable features implementation.
+
+Let's start at the begining of the pipline graph definition.
```

Comment:
This is what we should start with. Probably it could be "Section 1" (with "Section 0" describing the common words about hybrid CV/DL algorithms, video, pipelining, etc)